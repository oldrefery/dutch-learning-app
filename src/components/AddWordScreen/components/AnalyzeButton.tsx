import React, { useCallback } from 'react'
import { StyleSheet, View, Platform } from 'react-native'
import { GlassIconButton } from '@/components/glass/buttons/GlassIconButton'

interface AnalyzeButtonProps {
  isAnalyzing: boolean
  isCheckingDuplicate?: boolean
  onPress: () => void
  size?: 'small' | 'medium' | 'large'
}

/**
 * Analyze Button Component
 *
 * AI analysis button following HIG guidelines:
 * - Uses sparkles icon to indicate AI functionality
 * - Platform-specific icons (sparkles for iOS, sparkles-sharp for Android)
 * - Tinted glass button style
 * - Proper tap target (44x44pt)
 * - Disabled state during analysis
 *
 * @see https://developer.apple.com/design/human-interface-guidelines/buttons
 */
export function AnalyzeButton({
  isAnalyzing,
  isCheckingDuplicate = false,
  onPress,
  size = 'medium',
}: AnalyzeButtonProps) {
  const isLoading = isAnalyzing || isCheckingDuplicate

  const handlePress = useCallback(() => {
    if (!isLoading) {
      onPress()
    }
  }, [isLoading, onPress])

  // Use platform-specific sparkles icon
  const icon = Platform.OS === 'ios' ? 'sparkles' : 'sparkles-sharp'

  return (
    <View style={styles.container}>
      <GlassIconButton
        icon={icon}
        onPress={handlePress}
        variant="tinted"
        size={size}
        disabled={isLoading}
        accessibilityLabel="Analyze word with AI"
        accessibilityHint="Analyzes the Dutch word using artificial intelligence"
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {},
})
