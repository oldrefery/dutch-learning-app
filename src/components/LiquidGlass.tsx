import React from 'react'
import {
  View,
  StyleSheet,
  Platform,
  type StyleProp,
  type ViewStyle,
  useColorScheme,
} from 'react-native'
import { BlurView } from 'expo-blur'
import { usePreferReducedTransparency } from '@/hooks/usePreferReducedTransparency'
import {
  GlassDefaults,
  GlassOutline,
  LiquidGlassElevation,
  LiquidGlassIntensityMode,
  LiquidGlassRadius,
  LiquidGlassTint,
} from '@/constants/GlassConstants'

export type LiquidGlassProps = {
  children?: React.ReactNode
  tint?: LiquidGlassTint
  intensity?: LiquidGlassIntensityMode | number
  radius?: LiquidGlassRadius | number
  withOutline?: boolean
  padding?: number
  elevation?: LiquidGlassElevation
  testID?: string
  style?: StyleProp<ViewStyle>
}

export const LiquidGlass: React.FC<LiquidGlassProps> = ({
  children,
  tint = GlassDefaults.tint,
  intensity = GlassDefaults.intensityMode,
  radius = GlassDefaults.radius,
  withOutline = GlassDefaults.withOutline,
  padding,
  elevation = Platform.OS === 'android'
    ? GlassDefaults.elevationAndroid
    : LiquidGlassElevation.None,
  testID,
  style,
}) => {
  const colorScheme = useColorScheme()
  const isDarkMode = colorScheme === 'dark'
  const reduceTransparency = usePreferReducedTransparency()

  const resolvedTint: 'light' | 'dark' =
    tint === LiquidGlassTint.Auto
      ? isDarkMode
        ? 'dark'
        : 'light'
      : (tint as 'light' | 'dark')

  const resolvedIntensity: number =
    typeof intensity === 'number'
      ? intensity
      : intensity === LiquidGlassIntensityMode.Adaptive
        ? isDarkMode
          ? GlassDefaults.intensityDark
          : GlassDefaults.intensityLight
        : GlassDefaults.intensityLight

  const borderColor = isDarkMode ? GlassOutline.dark : GlassOutline.light
  const borderStyle = withOutline
    ? { borderColor, borderWidth: StyleSheet.hairlineWidth }
    : null

  const resolvedRadius = typeof radius === 'number' ? radius : Number(radius)

  const containerElevation =
    Platform.OS === 'android' && elevation && elevation > 0
      ? { elevation, shadowColor: '#000', shadowOpacity: 0.1, shadowRadius: 6 }
      : undefined

  const resolvedPadding = padding ?? 0

  return (
    <View style={[styles.container, containerElevation, style]} testID={testID}>
      {reduceTransparency ? (
        <View
          style={[
            styles.blur,
            { borderRadius: resolvedRadius },
            borderStyle,
            resolvedPadding ? { padding: resolvedPadding } : null,
            {
              backgroundColor: isDarkMode
                ? 'rgba(255,255,255,0.06)'
                : 'rgba(255,255,255,0.5)',
            },
          ]}
        >
          {children}
        </View>
      ) : (
        <BlurView
          intensity={resolvedIntensity}
          tint={resolvedTint}
          experimentalBlurMethod={'dimezisBlurView'}
          style={[
            styles.blur,
            { borderRadius: resolvedRadius },
            borderStyle,
            resolvedPadding ? { padding: resolvedPadding } : null,
          ]}
        >
          {children}
        </BlurView>
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    overflow: 'hidden',
  },
  blur: {
    width: '100%',
  },
})

export default LiquidGlass
