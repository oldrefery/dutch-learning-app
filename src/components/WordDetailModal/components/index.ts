export { default as WordTranslations } from './WordTranslations'
export { default as WordExamples } from './WordExamples'
export { default as WordProgress } from './WordProgress'
export { default as WordImage } from './WordImage'
export { default as WordStatus } from './WordStatus'
export { default as WordDetailHeader } from './WordDetailHeader'
export { default as WordPartOfSpeech } from './WordPartOfSpeech'
