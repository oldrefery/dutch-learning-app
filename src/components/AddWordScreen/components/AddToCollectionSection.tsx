import React from 'react'
import {
  TouchableOpacity,
  ActivityIndicator,
  useColorScheme,
} from 'react-native'
import { TextThemed, ViewThemed } from '@/components/Themed'
import CollectionSelector from '@/components/CollectionSelector'
import { Colors } from '@/constants/Colors'
import type { AddToCollectionSectionProps } from '../types/AddWordTypes'
import { addToCollectionStyles } from '../styles/AddToCollectionSection.styles'

export function AddToCollectionSection({
  selectedCollection,
  onCollectionSelect,
  onAddWord,
  isAdding,
}: AddToCollectionSectionProps) {
  const colorScheme = useColorScheme() ?? 'light'
  return (
    <ViewThemed
      style={[
        addToCollectionStyles.addToCollectionSection,
        {
          borderTopColor:
            colorScheme === 'dark'
              ? Colors.dark.backgroundSecondary
              : Colors.neutral[200],
        },
      ]}
    >
      <TextThemed style={addToCollectionStyles.addToCollectionTitle}>
        Add to Collection
      </TextThemed>

      <ViewThemed style={addToCollectionStyles.collectionSelectorContainer}>
        <TextThemed
          style={addToCollectionStyles.collectionLabel}
          lightColor={Colors.neutral[700]}
          darkColor={Colors.dark.textSecondary}
        >
          Select Collection:
        </TextThemed>
        <CollectionSelector
          selectedCollectionId={selectedCollection?.collection_id || null}
          onCollectionSelect={onCollectionSelect}
        />
      </ViewThemed>

      <TouchableOpacity
        style={[
          addToCollectionStyles.addButton,
          {
            backgroundColor:
              colorScheme === 'dark'
                ? Colors.dark.tint
                : Colors.primary.DEFAULT,
          },
          isAdding && addToCollectionStyles.addButtonDisabled,
        ]}
        onPress={onAddWord}
        disabled={isAdding || !selectedCollection}
      >
        {isAdding ? (
          <ActivityIndicator size="small" color="white" />
        ) : (
          <TextThemed style={addToCollectionStyles.addButtonText}>
            Add Word
          </TextThemed>
        )}
      </TouchableOpacity>
    </ViewThemed>
  )
}
