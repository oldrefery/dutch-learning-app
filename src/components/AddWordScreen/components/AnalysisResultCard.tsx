import React from 'react'
import { ScrollView, TouchableOpacity, Image } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import { TextThemed, ViewThemed } from '@/components/Themed'
import { SelectableText } from '@/components/SelectableText'
import { Colors } from '@/constants/Colors'
import { CopyButton } from '@/components/CopyButton'
import { GlassCapsuleButton } from '@/components/glass/buttons'
import { formatAnalysisResultForCopying } from '@/utils/wordTextFormatter'
import type { AnalysisResultCardProps } from '../types/AddWordTypes'
import { analysisResultStyles } from '../styles/AnalysisResultCard.styles'

export function AnalysisResultCard({
  analysisResult,
  isPlayingAudio,
  onPlayPronunciation,
  onShowImageSelector,
  isAlreadyInCollection = false,
  isCheckingDuplicate = false,
}: AnalysisResultCardProps) {
  return (
    <ScrollView style={analysisResultStyles.resultContainer}>
      <ViewThemed style={analysisResultStyles.resultCard}>
        <ViewThemed style={analysisResultStyles.titleContainer}>
          <TextThemed style={analysisResultStyles.resultTitle}>
            Analysis Result
          </TextThemed>
          <ViewThemed style={analysisResultStyles.titleActions}>
            <CopyButton text={formatAnalysisResultForCopying(analysisResult)} />
            {isCheckingDuplicate && (
              <ViewThemed style={analysisResultStyles.checkingBadge}>
                <Ionicons
                  name="hourglass"
                  size={16}
                  color={Colors.neutral[500]}
                />
                <TextThemed style={analysisResultStyles.checkingText}>
                  Checking...
                </TextThemed>
              </ViewThemed>
            )}
            {isAlreadyInCollection && !isCheckingDuplicate && (
              <ViewThemed style={analysisResultStyles.alreadyExistsBadge}>
                <Ionicons
                  name="checkmark-circle"
                  size={16}
                  color={Colors.success.DEFAULT}
                />
                <TextThemed style={analysisResultStyles.alreadyExistsText}>
                  Already in collection
                </TextThemed>
              </ViewThemed>
            )}
          </ViewThemed>
        </ViewThemed>

        <ViewThemed style={analysisResultStyles.resultRow}>
          <TextThemed style={analysisResultStyles.resultLabel}>
            Word:
          </TextThemed>
          <ViewThemed style={analysisResultStyles.wordWithPronunciation}>
            <TextThemed style={analysisResultStyles.resultValue} selectable>
              {analysisResult.dutch_lemma}
            </TextThemed>
            {analysisResult.tts_url && (
              <TouchableOpacity
                style={analysisResultStyles.pronunciationButton}
                onPress={() => onPlayPronunciation(analysisResult.tts_url!)}
                disabled={isPlayingAudio}
              >
                <Ionicons
                  name={isPlayingAudio ? 'volume-high' : 'volume-medium'}
                  size={20}
                  color={Colors.primary.DEFAULT}
                />
              </TouchableOpacity>
            )}
          </ViewThemed>
        </ViewThemed>

        <ViewThemed style={analysisResultStyles.resultRow}>
          <TextThemed style={analysisResultStyles.resultLabel}>
            Type:
          </TextThemed>
          <TextThemed style={analysisResultStyles.resultValue} selectable>
            {analysisResult.part_of_speech}
            {analysisResult.is_irregular ? ' (irregular)' : ''}
            {analysisResult.article ? ` (${analysisResult.article})` : ''}
            {analysisResult.is_reflexive ? ' (reflexive)' : ''}
            {analysisResult.is_expression
              ? ` (${analysisResult.expression_type || 'expression'})`
              : ''}
            {analysisResult.is_separable ? ' (separable)' : ''}
          </TextThemed>
        </ViewThemed>

        {analysisResult.is_separable &&
          analysisResult.prefix_part &&
          analysisResult.root_verb && (
            <ViewThemed style={analysisResultStyles.resultRow}>
              <TextThemed style={analysisResultStyles.resultLabel}>
                Parts:
              </TextThemed>
              <TextThemed style={analysisResultStyles.resultValue} selectable>
                <TextThemed style={analysisResultStyles.prefixText}>
                  {analysisResult.prefix_part}
                </TextThemed>{' '}
                + {analysisResult.root_verb}
              </TextThemed>
            </ViewThemed>
          )}

        <ViewThemed style={analysisResultStyles.resultSection}>
          <TextThemed style={analysisResultStyles.resultLabel}>
            English:
          </TextThemed>
          {analysisResult.translations.en.map((translation, index) => (
            <SelectableText
              key={index}
              style={analysisResultStyles.translationText}
              copyText={translation}
            >
              • {translation}
            </SelectableText>
          ))}
        </ViewThemed>

        {analysisResult.translations.ru && (
          <ViewThemed style={analysisResultStyles.resultSection}>
            <TextThemed style={analysisResultStyles.resultLabel}>
              Russian:
            </TextThemed>
            {analysisResult.translations.ru.map((translation, index) => (
              <SelectableText
                key={index}
                style={analysisResultStyles.translationText}
                copyText={translation}
              >
                • {translation}
              </SelectableText>
            ))}
          </ViewThemed>
        )}

        {analysisResult.image_url && (
          <ViewThemed style={analysisResultStyles.resultSection}>
            <TextThemed style={analysisResultStyles.resultLabel}>
              Visual:
            </TextThemed>
            <ViewThemed style={analysisResultStyles.imageContainer}>
              <Image
                source={{ uri: analysisResult.image_url }}
                style={analysisResultStyles.associationImage}
                resizeMode="cover"
              />
              <GlassCapsuleButton
                icon="images"
                text="Change Image"
                onPress={onShowImageSelector}
                variant="tinted"
                size="medium"
                accessibilityLabel="Change image"
                accessibilityHint="Opens image selector to choose a new image"
              />
            </ViewThemed>
          </ViewThemed>
        )}

        {analysisResult.examples && analysisResult.examples.length > 0 && (
          <ViewThemed style={analysisResultStyles.resultSection}>
            <TextThemed style={analysisResultStyles.resultLabel}>
              Examples:
            </TextThemed>
            {analysisResult.examples.map((example, index) => (
              <ViewThemed key={index} style={analysisResultStyles.exampleCard}>
                <SelectableText
                  style={analysisResultStyles.exampleDutch}
                  copyText={example.nl}
                >
                  {example.nl}
                </SelectableText>
                <SelectableText
                  style={analysisResultStyles.exampleTranslation}
                  copyText={example.en}
                >
                  {example.en}
                </SelectableText>
                {example.ru && (
                  <SelectableText
                    style={analysisResultStyles.exampleTranslation}
                    copyText={example.ru}
                  >
                    {example.ru}
                  </SelectableText>
                )}
              </ViewThemed>
            ))}
          </ViewThemed>
        )}
      </ViewThemed>
    </ScrollView>
  )
}
