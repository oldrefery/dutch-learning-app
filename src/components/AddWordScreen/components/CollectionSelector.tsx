import React from 'react'
import { TouchableOpacity, StyleSheet } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import { TextThemed } from '@/components/Themed'
import { Colors } from '@/constants/Colors'
import type { Collection } from '@/types/database'

interface CollectionSelectorProps {
  selectedCollection: Collection | null
  onPress: () => void
  colorScheme: 'light' | 'dark'
}

/**
 * Collection Selector Component
 *
 * Displays the currently selected collection and allows changing it.
 * Used in compact word input forms.
 */
export function CollectionSelector({
  selectedCollection,
  onPress,
  colorScheme,
}: CollectionSelectorProps) {
  return (
    <TouchableOpacity onPress={onPress} style={styles.collectionRow}>
      <TextThemed
        style={[
          styles.label,
          {
            color:
              colorScheme === 'dark'
                ? Colors.dark.textSecondary
                : Colors.neutral[600],
          },
        ]}
      >
        Adding to:{' '}
      </TextThemed>
      <TextThemed
        style={[
          styles.collectionName,
          {
            color:
              colorScheme === 'dark'
                ? Colors.dark.tint
                : Colors.primary.DEFAULT,
          },
        ]}
      >
        {selectedCollection?.name || 'Select Collection'}
      </TextThemed>
      <Ionicons
        name="chevron-down"
        size={16}
        color={
          colorScheme === 'dark'
            ? Colors.dark.textSecondary
            : Colors.neutral[600]
        }
      />
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  collectionRow: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
    marginBottom: 8,
    backgroundColor: 'transparent',
  },
  label: {
    fontSize: 16,
    fontWeight: '500',
  },
  collectionName: {
    fontSize: 16,
    fontWeight: '600',
    flex: 1,
  },
})
