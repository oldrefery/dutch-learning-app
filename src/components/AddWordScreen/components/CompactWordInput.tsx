import React from 'react'
import {
  TextInput,
  TouchableOpacity,
  ActivityIndicator,
  useColorScheme,
  ActionSheetIOS,
  Platform,
} from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import { TextThemed, ViewThemed } from '@/components/Themed'
import { Colors } from '@/constants/Colors'
import type { Collection } from '@/types/database'

interface CompactWordInputProps {
  inputWord: string
  setInputWord: (word: string) => void
  onAnalyze: () => void
  isAnalyzing: boolean
  isCheckingDuplicate: boolean
  selectedCollection: Collection | null
  collections: Collection[]
  onCollectionSelect: (collection: Collection | null) => void
  onCancel?: () => void
}

// Styles helper to reduce complexity
const getStyles = (colorScheme: 'light' | 'dark') => ({
  container: {
    paddingHorizontal: 16,
    paddingVertical: 12,
    backgroundColor:
      colorScheme === 'dark'
        ? Colors.dark.background
        : Colors.background.primary,
  },
  collectionRow: {
    flexDirection: 'row' as const,
    alignItems: 'center' as const,
    paddingVertical: 8,
    marginBottom: 8,
  },
  inputContainer: {
    flexDirection: 'row' as const,
    alignItems: 'center' as const,
    backgroundColor:
      colorScheme === 'dark'
        ? Colors.dark.backgroundSecondary
        : Colors.neutral[50],
    borderRadius: 8,
    borderWidth: 1,
    borderColor:
      colorScheme === 'dark'
        ? Colors.dark.backgroundTertiary
        : Colors.neutral[200],
  },
  loadingContainer: {
    flexDirection: 'row' as const,
    alignItems: 'center' as const,
    marginTop: 8,
    paddingVertical: 4,
  },
})

const CollectionSelector = ({
  selectedCollection,
  onPress,
  colorScheme,
}: {
  selectedCollection: Collection | null
  onPress: () => void
  colorScheme: 'light' | 'dark'
}) => (
  <TouchableOpacity
    onPress={onPress}
    style={getStyles(colorScheme).collectionRow}
  >
    <TextThemed
      style={{
        fontSize: 16,
        fontWeight: '500',
        color:
          colorScheme === 'dark'
            ? Colors.dark.textSecondary
            : Colors.neutral[600],
      }}
    >
      Adding to:{' '}
    </TextThemed>
    <TextThemed
      style={{
        fontSize: 16,
        fontWeight: '600',
        color:
          colorScheme === 'dark' ? Colors.dark.tint : Colors.primary.DEFAULT,
        flex: 1,
      }}
    >
      {selectedCollection?.name || 'Select Collection'}
    </TextThemed>
    <Ionicons
      name="chevron-down"
      size={16}
      color={
        colorScheme === 'dark' ? Colors.dark.textSecondary : Colors.neutral[600]
      }
    />
  </TouchableOpacity>
)

const AnalyzeButton = ({
  isAnalyzing,
  isCheckingDuplicate,
  onAnalyze,
  colorScheme,
}: {
  isAnalyzing: boolean
  isCheckingDuplicate: boolean
  onAnalyze: () => void
  colorScheme: 'light' | 'dark'
}) => (
  <TouchableOpacity
    style={{
      backgroundColor:
        colorScheme === 'dark' ? Colors.dark.tint : Colors.primary.DEFAULT,
      paddingHorizontal: 16,
      paddingVertical: 8,
      marginRight: 4,
      borderRadius: 6,
    }}
    onPress={onAnalyze}
    disabled={isAnalyzing || isCheckingDuplicate}
  >
    {isAnalyzing || isCheckingDuplicate ? (
      <ActivityIndicator size="small" color="white" />
    ) : (
      <Ionicons name="search" size={16} color="white" />
    )}
  </TouchableOpacity>
)

export function CompactWordInput({
  inputWord,
  setInputWord,
  onAnalyze,
  isAnalyzing,
  isCheckingDuplicate,
  selectedCollection,
  collections,
  onCollectionSelect,
  onCancel,
}: CompactWordInputProps) {
  const colorScheme = useColorScheme() ?? 'light'
  const styles = getStyles(colorScheme)

  const handleCollectionPress = () => {
    if (Platform.OS === 'ios') {
      const options = collections.map(c => c.name)
      options.push('Cancel')

      ActionSheetIOS.showActionSheetWithOptions(
        {
          options,
          cancelButtonIndex: options.length - 1,
          title: 'Select Collection',
        },
        buttonIndex => {
          if (buttonIndex < collections.length) {
            onCollectionSelect(collections[buttonIndex])
          }
        }
      )
    }
  }

  return (
    <ViewThemed style={styles.container}>
      <CollectionSelector
        selectedCollection={selectedCollection}
        onPress={handleCollectionPress}
        colorScheme={colorScheme}
      />

      <ViewThemed style={styles.inputContainer}>
        <Ionicons
          name="search"
          size={20}
          color={
            colorScheme === 'dark'
              ? Colors.dark.textSecondary
              : Colors.neutral[400]
          }
          style={{ marginLeft: 12 }}
        />

        <TextInput
          style={{
            flex: 1,
            paddingVertical: 12,
            paddingHorizontal: 8,
            fontSize: 16,
            color:
              colorScheme === 'dark' ? Colors.dark.text : Colors.neutral[900],
          }}
          value={inputWord}
          onChangeText={setInputWord}
          placeholder="Enter Dutch word..."
          placeholderTextColor={
            colorScheme === 'dark'
              ? Colors.dark.textTertiary
              : Colors.neutral[400]
          }
          autoCapitalize="none"
          autoCorrect={false}
          autoComplete="off"
          textContentType="none"
          spellCheck={false}
          keyboardType="ascii-capable"
          returnKeyType="search"
          onSubmitEditing={onAnalyze}
        />

        <AnalyzeButton
          isAnalyzing={isAnalyzing}
          isCheckingDuplicate={isCheckingDuplicate}
          onAnalyze={onAnalyze}
          colorScheme={colorScheme}
        />
      </ViewThemed>

      {(isAnalyzing || isCheckingDuplicate) && (
        <ViewThemed style={styles.loadingContainer}>
          <ActivityIndicator size="small" color={Colors.primary.DEFAULT} />
          <TextThemed
            style={{
              marginLeft: 8,
              fontSize: 14,
              color:
                colorScheme === 'dark'
                  ? Colors.dark.textSecondary
                  : Colors.neutral[600],
            }}
          >
            {isCheckingDuplicate
              ? 'Checking for duplicates...'
              : 'Analyzing word with AI...'}
          </TextThemed>
        </ViewThemed>
      )}
    </ViewThemed>
  )
}
