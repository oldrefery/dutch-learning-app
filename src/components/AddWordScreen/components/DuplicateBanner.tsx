import React from 'react'
import { TouchableOpacity, useColorScheme, StyleSheet } from 'react-native'
import { Ionicons } from '@expo/vector-icons'
import { router } from 'expo-router'
import { TextThemed, ViewThemed } from '@/components/Themed'
import { Colors } from '@/constants/Colors'
import { ROUTES } from '@/constants/Routes'
import type { Collection } from '@/types/database'

interface DuplicateWordData {
  word_id: string
  dutch_lemma: string
  collection_id: string
  part_of_speech: string
  article?: string
}

interface DuplicateBannerProps {
  duplicateWord: DuplicateWordData
  collections: Collection[]
  onNavigateToCollection?: () => void
}

export function DuplicateBanner({
  duplicateWord,
  collections,
  onNavigateToCollection,
}: DuplicateBannerProps) {
  const colorScheme = useColorScheme() ?? 'light'

  const collection = collections.find(
    c => c.collection_id === duplicateWord.collection_id
  )

  const handleNavigateToCollection = () => {
    onNavigateToCollection?.()
    const collectionPath = ROUTES.COLLECTION_DETAIL(duplicateWord.collection_id)
    router.push({
      pathname: collectionPath,
      params: { highlightWordId: duplicateWord.word_id },
    })
  }

  return (
    <ViewThemed
      style={[
        styles.container,
        {
          backgroundColor:
            colorScheme === 'dark'
              ? Colors.dark.backgroundSecondary
              : Colors.background.secondary,
          borderColor:
            colorScheme === 'dark' ? Colors.dark.tint : Colors.primary.DEFAULT,
          borderWidth: 2,
        },
      ]}
    >
      <ViewThemed style={styles.content}>
        <Ionicons
          name="information-circle"
          size={20}
          color={
            colorScheme === 'dark' ? Colors.dark.tint : Colors.primary.DEFAULT
          }
        />
        <TextThemed
          style={[
            styles.text,
            {
              color:
                colorScheme === 'dark' ? Colors.dark.text : Colors.neutral[800],
            },
          ]}
        >
          Already in{' '}
        </TextThemed>
        <TouchableOpacity
          onPress={handleNavigateToCollection}
          activeOpacity={0.7}
        >
          <TextThemed
            style={[
              styles.collectionName,
              {
                color:
                  colorScheme === 'dark'
                    ? Colors.dark.tint
                    : Colors.primary.DEFAULT,
              },
            ]}
          >
            {collection?.name || 'Collection'}
          </TextThemed>
        </TouchableOpacity>
      </ViewThemed>
    </ViewThemed>
  )
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 12,
    padding: 16,
    marginHorizontal: 16,
    marginBottom: 8,
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    fontSize: 16,
    fontWeight: '500',
    marginLeft: 8,
  },
  collectionName: {
    fontSize: 16,
    fontWeight: '600',
    textDecorationLine: 'underline',
  },
})
