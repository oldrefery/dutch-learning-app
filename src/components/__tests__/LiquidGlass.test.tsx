import React from 'react'
import renderer from 'react-test-renderer'
import { LiquidGlass } from '@/components/LiquidGlass'

describe('LiquidGlass', () => {
  it('renders with defaults', () => {
    const tree = renderer.create(<LiquidGlass />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with custom radius and no outline', () => {
    const tree = renderer
      .create(<LiquidGlass radius={16} withOutline={false} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
