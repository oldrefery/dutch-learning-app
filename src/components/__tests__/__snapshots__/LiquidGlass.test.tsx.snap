// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiquidGlass renders with custom radius and no outline 1`] = `null`;

exports[`LiquidGlass renders with defaults 1`] = `null`;
